### ISSUE TITLE
Error handling

### ISSUE THREAD

**Issue by eoriont** [C0BBL3/space-empires#34]
In all of your engines, when you validate inputs, you only run the moves if they are valid, but you don't make sure the game knows the player is cheating. 

In general, aggressive input validation is a very good practice, because it allows you to stop thinking of cases where your data is null. See these articles for more information:

https://blog.securityinnovation.com/blog/2010/12/constrain-reject-and-sanitize-input.html
https://mne.tools/0.15/auto_tutorials/plot_artifacts_correction_rejection.html

It looks like you implemented it in your movement engine, but then commented it out. All you'd have to do is uncomment that, and add an else statement to lines 39 and 59 in your economic engine, and some other lines in your combat engine. It's a simple fix, really.

This will allow your debugging to be much easier. Plus, it will guarantee that unexpected behavior is the strategy's fault!

(Don't close this without fixing)

**Comment by C0BBL3** [comment]
You're wrong, see lines in 17 and 21-24 in the movement engine, and in the economic engine see lines 41, 60, and 70. (also GitHub isn't working so this is local rn but I do check if they cheat or not)
